From: Shengjing Zhu <zhsj@debian.org>
Date: Tue, 26 Mar 2019 15:49:40 +0800
Subject: Fix UUID struct field not encoded in msgpack

Upstream uses the old codec library, so there's no problem.
Bug-Debian: https://bugs.debian.org/923866
Forwarded: no
---
 deb/local.go    | 4 ++--
 deb/snapshot.go | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/deb/local.go b/deb/local.go
index 229f6e1..5b8eb2a 100644
--- a/deb/local.go
+++ b/deb/local.go
@@ -15,7 +15,7 @@ import (
 // LocalRepo is a collection of packages created locally
 type LocalRepo struct {
 	// Permanent internal ID
-	UUID string `json:"-"`
+	UUID string `codec:"UUID" json:"-"`
 	// User-assigned name
 	Name string
 	// Comment
@@ -25,7 +25,7 @@ type LocalRepo struct {
 	// DefaultComponent
 	DefaultComponent string `codec:",omitempty"`
 	// Uploaders configuration
-	Uploaders *Uploaders `code:",omitempty" json:"-"`
+	Uploaders *Uploaders `codec:"Uploaders,omitempty" json:"-"`
 	// "Snapshot" of current list of packages
 	packageRefs *PackageRefList
 }
diff --git a/deb/snapshot.go b/deb/snapshot.go
index 307433a..aef19ae 100644
--- a/deb/snapshot.go
+++ b/deb/snapshot.go
@@ -19,15 +19,15 @@ import (
 // Snapshot is immutable state of repository: list of packages
 type Snapshot struct {
 	// Persisten internal ID
-	UUID string `json:"-"`
+	UUID string `codec:"UUID" json:"-"`
 	// Human-readable name
 	Name string
 	// Date of creation
 	CreatedAt time.Time
 
 	// Source: kind + ID
-	SourceKind string   `json:"-"`
-	SourceIDs  []string `json:"-"`
+	SourceKind string   `codec:"SourceKind" json:"-"`
+	SourceIDs  []string `codec:"SourceIDs" json:"-"`
 	// Description of how snapshot was created
 	Description string
 
